src/cpp/core/include/core/system/ProcessArgs.hpp

```diff
@@ -81,8 +81,8 @@ class ProcessArgs : boost::noncopyable
       {
          for (std::size_t i = 0; i<argCount_; ++i)
             delete [] args_[i] ;
-         delete [] args_ ;
       }
+      delete [] args_ ;
```

Comment:
This doesn't look right either -- the constructor calls `setArgs()`, which also calls `freeArgs()`, which means we might attempt to delete a NULL pointer. We should guard the deletion.